lib/Runtime/ByteCode/ByteCodeWriter.h

```diff
@@ -96,6 +96,28 @@ namespace Js
 
             template <LayoutSize layoutSize> void EncodeOpCode(uint16 op, ByteCodeWriter* writer);
             template <> void EncodeOpCode<SmallLayout>(uint16 op, ByteCodeWriter* writer);
+
+#define     DEFINE_TEMPLATE_BYTECODEWRITER_DATA_ENCODEOPCODE_FNC                   \
+            template <>                                                            \
+            inline void ByteCodeWriter::Data::EncodeOpCode<SmallLayout>(uint16 op, \
+                                                           ByteCodeWriter* writer) \
+            {                                                                      \
+                DebugOnly(const uint offset = currentOffset);                      \
+                if (op <= (uint16)Js::OpCode::MaxByteSizedOpcodes)                 \
+                {                                                                  \
+                    byte byteop = (byte)op;                                        \
+                    Write(&byteop, sizeof(byte));                                  \
+                }                                                                  \
+                else                                                               \
+                {                                                                  \
+                    byte byteop = (byte)Js::OpCode::ExtendedOpcodePrefix;          \
+                    Write(&byteop, sizeof(byte));                                  \
+                    Write(&op, sizeof(uint16));                                    \
+                }                                                                  \
+                Assert(OpCodeUtil::EncodedSize((Js::OpCode)op, SmallLayout)        \
+                       == (currentOffset - offset));                               \
+            }
```

Comment:
Indeed weird. Tried other things [i.e. remove inline / use it from header ].. Older Clang behaves strange :/ Actually, we already define those in multiple places (using `include` )